### ISSUE TITLE
[DataForm] Issue while setting up groups from the metadata

### ISSUE THREAD

**Issue by tsonevn** [telerik/nativescript-ui-feedback#216]
Setting up groupName in the JSON object will not be respected and the  DataForm fields will not be grouped. This issue is reproducible fo both iOS and Android. I am attaching sample project, which could be used for debugging.
[Archive.zip](https://github.com/telerik/nativescript-ui-feedback/files/1084251/Archive.zip)

**Comment by RobertGardner** [comment]
It looks like this is fixed in NS UI 3.0.0, but it looks like the non-grouped properties all have to show after the grouped properties. It seems they should show up in their index order.

Ideally, I'd add an index to the `<TKEntityProperty>` tag and one to the `<TKPropertyGroup>` tag, and those would be used to order the properties on the form. I could then easily sprinkle the ungrouped properties before, after, or between the groups.

**Comment by tgpetrov** [comment]
Hi guys,

The existing issue is that giving a `groupName` to the properties through the JSON metadata will not cause the DataForm to show groups:
```
{
  "propertyAnnotations":
  [
    {
      "name": "name",
      "groupName": "Group1"
    },
    {
      "name": "age",
      "groupName": "Group1"
    },
    {
      "name": "email",
      "groupName": "Group2"
    },
    {
      "name": "city",
      "groupName": "Group2"
    }
  ]
}
```
The workaround is to also define the desired groups through html (or xml):
```
<RadDataForm [source]="person" [metadata]="personMetadata">
    <TKPropertyGroup tkDataFormGroups name="Group1">
        <TKEntityProperty tkPropertyGroupProperties name="name"></TKEntityProperty>
        <TKEntityProperty tkPropertyGroupProperties name="age"></TKEntityProperty>
    </TKPropertyGroup>
    <TKPropertyGroup tkDataFormGroups name="Group2">
        <TKEntityProperty tkPropertyGroupProperties name="email"></TKEntityProperty>
        <TKEntityProperty tkPropertyGroupProperties name="city"></TKEntityProperty>
    </TKPropertyGroup>
</RadDataForm>
```
@RobertGardner Currently the indexes are considered only within the group meaning that the indexes of the non-grouped properties will have effect only among them but will not place them before any groups.
We will consider your suggestion and work on providing a way to give you full control on the order of all properties and groups.

**Comment by DeonNel1** [comment]
Any update on this issue?  I have a scenario where I have to generate dynamic grouped templates and this is not possible if it has to be explicitly defined in the xml...

**Comment by Shantanu-Kotambkar** [comment]
@tsonevn Need an update on this

**Comment by zbranzov** [comment]
The fix is released in nativescript-ui-dataform@3.9.0

**Comment by tgpetrov** [comment]
@RobertGardner, yes, as noted [here](https://docs.telerik.com/devtools/nativescript-ui/Controls/Angular/DataForm/Groups/dataform-groups-overview#add-groups-with-json), there is a limitation in the current implementation that when the groups are setup through json, they can't be accessed through `getGroupByName`. In order to apply customisations you can use the `groupUpdate` event, get the native group and apply a [GroupTitleStyle](https://docs.telerik.com/devtools/nativescript-ui/api/classes/grouptitlestyle.html). Here's an example:
```
public onGroupUpdate(args) {
    let nativeGroup = args.group;
    let style = new GroupTitleStyle();
    style.labelTextColor = "#ff0000";
    (<any>this.myRuntimeDataFormComp.dataForm)._applyGroupTitleStyle(nativeGroup, style);
}
```
